server/src/main/java/org/opensearch/search/pipeline/SearchPipelineService.java

```diff
@@ -354,13 +358,28 @@ public PipelinedRequest resolvePipeline(SearchRequest searchRequest) throws Exce
             } catch (Exception e) {
                 throw new SearchPipelineProcessingException(e);
             }
-        } else if (searchRequest.pipeline() != null) {
-            String pipelineId = searchRequest.pipeline();
-            PipelineHolder pipelineHolder = pipelines.get(pipelineId);
-            if (pipelineHolder == null) {
-                throw new IllegalArgumentException("Pipeline " + pipelineId + " is not defined");
+        } else {
+            String pipelineId = NOOP_PIPELINE_ID;
+            if (searchRequest.pipeline() != null) {
+                // Named pipeline specified for the request
+                pipelineId = searchRequest.pipeline();
+            } else if (searchRequest.indices() != null && searchRequest.indices().length == 1) {
```

Comment:
Sorry, just to be sure, would this check properly handle `*` or `_all` index placeholders?

Reply:
Hmm... that's a good question. I'll check.

Reply:
Looking into it, it seems that the check will return true for requests like `/*/_search` and `/_all/_search`, but that the value of the index used is `*` or `_all`, which doesn't match anything in the IndexMetadata.

So, ultimately we get the "correct" behavior (for now) of not applying a search pipeline on a wildcard.

I'm going to create a follow-up issue to address more complicated questions around multi-index searches, aliases, and remote indexes.

Reply:
https://github.com/opensearch-project/OpenSearch/issues/7512

Reply:
Thanks @msfroh !